client/src/components/SignUp/SignUp.js

```diff
@@ -1,101 +1,61 @@
+    const user = useSelector((state) => state.user);
+    const dispatch = useDispatch();
 
-    const errorToast = (message) => {
-        toast.error(message, {
-            position: "top-center",
-            autoClose: 3000,
-            closeOnClick: true,
-            hideProgressBar: true,
-            pauseOnHover: true,
-            draggable: true,
-            progress: undefined
-        });
-    };
-    //TOAST TO DISPLAY FOR SUCCESSFULL SIGNUP
-    const successToast = (message) => {
-        toast.success(message, {
-            position: "top-right",
-            autoClose: 2000,
-            hideProgressBar: true,
-            closeOnClick: true,
-            pauseOnHover: true,
-            draggable: true,
-            progress: undefined
-        });
-    };
+    const userSignup = (e) => {
+        e.preventDefault();
 
-    //FUNCTION TO DO APPROPRIATE TASK ON CLICKING SUBMIT BUTTON
-    const PostData = () => {
-        //CONDITIONS TO CHECK VALID INPUT DETAILS
-        if (formData.name !== "") {
-            if (formData.email !== "") {
-                // eslint-disable-next-line
-                const re = /^(([^<>()[\]\\.,;:"\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
-                if (re.test(formData.email)) {
-                    if (formData.password !== "") {
-                        if (formData.course !== "") {
-                            if (formData.year !== "") {
-                                if (formData.college !== "") {
-                                    // If signup process is successfull via an api then
-                                    // we will show a toast and redirect the user in a .then() block
-                                    successToast("Signup successfull");
-                                    history.push("/");
-                                } else {
-                                    errorToast("Please enter college name");
-                                }
-                            } else {
-                                errorToast("Please enter year");
-                            }
-                        } else {
-                            errorToast("Please enter course name");
-                        }
-                    } else {
-                        errorToast("Please enter password");
-                    }
-                } else {
-                    errorToast("Invalid email id");
-                }
-            } else {
-                errorToast("Please enter email");
-            }
-        } else {
-            errorToast("Please enter name");
-        }
+        const user = {
+            firstName,
+            lastName,
+            userName,
+            email,
+            password,
+            course,
+            year,
+            collegeName,
+            contactNumber
+        };
+        dispatch(signup(user));
```

Comment:
You can directly pass value rather than create object
ex `dispatch(signup({firstname, lastname...}));`